### PR TITLE
Storing Tiny Items in your Mouth - The Cavity Search Update

### PR DESCRIPTION
**Pull request by GreatBigFailure**
Now, let's all try to be mature adults. No low hanging fruit. No giggling in the back of the class. No crude humor. We are a respectable, professional server for respectable, professional adults. 

**Shove that little disk right in your mouth / Why**

You are Captain Chokem, you are securing the disk. While normally disks would fit snuggly in a clown's ass, today you're feeling a little greedy. You want that disk yourself. You want that disk right in your mouth. So you go to put that disk in your mouth and- oh no! You can't! You even tried biting it, and disks usually don't really like teeth, but no dice. Thankfully this PR changes that.

**What**

This PR lets you smuggle tiny (and only tiny) items in your mouth. So the disk, the spare, syringes, etc. You can later spit the item into your hands, the floor, or at something in front of you if you're a gross jerk. 

**How to eat**
You can put an item in your mouth by biting an item that does not already have a bite interaction. So no smuggling pills or burgers in your cheeks. Those just get eaten. This also means anything that prevents biting, like most masks, will prevent you from putting disks in your mouth. 

**How to spit**
You can remove an item from your mouth by:
- Using the "spit out" verb
- Clicking the screen icon in the top left (same place as toggling rig helmets)
- Trying to put another item in your mouth when you already have one
- Sneezing
- Vomiting
- Getting punched in the stomach
- Being beheaded
- Speaking

An item being covered in glue prevents you spitting it out until it is unglued. Please make sure not to vomit with a glued item in your mouth.

**Eating glass**
I've also included an easy way to add interactions for specific things being smuggled in your mouth. This PR includes one for both glass shards and syringes, hint: don't. More to come later.

### To Do

- [ ] Actually make it work. I think something with equipping to the mouth slot is busted
- [x] Make it so you can't eat while something is in your mouth
- [ ] Maybe add mutation interactions
- [ ] Maybe add a verb/screen button to put items in your mouth, not just rely on the bite option?
- [ ] Clean up some things. Remove test chat lines.
- [x] Add examine text saying "They very clearly have something in their mouth" similar to examining someone with "a braindead stare"
- [x] Add a method to force someone to spit something out that isn't punching
- [x] Add checks for sharpness flags to add "realism" for putting a screwdriver or knife in your mouth
- [x] Make getting tazed, teeth knocked out, stunned in general, and coughing cause you to spit the item out

**Drafting this because it doesn't work yet. I also feel like this one might be controversial so maybe it can be argued away before I invest more time.**
**Please make suggestions/voice concerns. I'll try to incorporate things while I'm still focused on this**

[wip]

:cl:
 * rscadd: Spessmen can now shove tiny objects in their mouth and spit them up later.

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
I feel the need to mention this was born purely from my desire to hide the spare or other contraband while being searched by sec. That is the only inspiration. I swear.

**Comment by adacovsk** [comment]
I feel a punch in the head would also make you spit the item out, but it seems like you covered any possible exploits or pitfalls, except for the "eat everything" mutation

**Comment by GreatBigFailure** [comment]
> I feel a punch in the head would also make you spit the item out, but it seems like you covered any possible exploits or pitfalls

I might just make receiving any damage have a chance to do it, along with being tazed/stunned. But make punches to the gut a more guaranteed method. Since this is an inconvenience to secoffs I feel like gifting them with an IC excuse to punch assistants in the stomach is a good peace offering.

**Comment by Hinaichigo** [comment]
Instead of speaking making the item fall out, what about having it make the speech all mumbly, like trying to talk with one's mouth full? Could be funny.
Also maybe then you could cram rags into peoples' mouth's after cuffbuckling them to a chair to gag them.

Maybe the mouth could be made a storage/inventory slot instead of having it go through the bite mechanic.

**Comment by GreatBigFailure** [comment]
@Hinaichigo It's been suggested I change the speaking interaction to what you recommended, more like "offghicher theres nughfin in my mouff". Then have a percentage chance to spit the item out or drool while doing it. I think I prefer that suggestion and I'll likely change it to that. Thanks.

I doubt I'll add a way for other people to put items in your mouth in this PR. That's a bit more complicated and also people will make enough jokes as is.

**Comment by Hinaichigo** [comment]
> I doubt I'll add a way for other people to put items in your mouth in this PR. That's a bit more complicated and also people will make enough jokes as is.

Yeah actually it does open various cans of worms. Like.. what prevents you from spitting the gag out, and we already have mask-slot muzzles.

**Comment by SonixApache** [comment]
i personally cannot wait to be elbow-deep into my waifu's throat for the mandatory cavity searches

**Comment by gurfan** [comment]
I think you overestimate how big the average mouth really is.

**Comment by Eneocho** [comment]
What happens if you have something in your mouth and a mask on? Can you spit it out or is it trapped there until the mask is removed?

**Comment by GreatBigFailure** [comment]
> What happens if you have something in your mouth and a mask on? Can you spit it out or is it trapped there until the mask is removed?

You can spit it out. It's something I'll change later since I recognize the consistency issue. Still, I expect some unforeseen exploits happening with this and so I'd prefer anything that potentially worsens that to be added after some live testing.
Besides, you can vomit/eat with a mask on so this uses the same logic.

**Comment by Eneocho** [comment]
I'd flat out remove the mask check for consistency with eating and vomiting then.

**Comment by GreatBigFailure** [comment]
Fantastic, less work for me!

**Comment by WindowsErrors** [comment]
how do i store pills in my mouth like the mouse from mouse and the motorcycle or whatever name the book had

**Comment by Anton-Kr** [comment]
There should be a way to check person's mouth without punching them in the stomach.

**Comment by Eneocho** [comment]
Crowbar to the mouth

**Comment by Hinaichigo** [comment]
> There should be a way to check person's mouth without punching them in the stomach.

Empty handed disarm targeting the mouth maybe?

**Comment by GreatBigFailure** [comment]
I'd appreciate those voting against it to voice their concerns. I can always change things.

>There should be a way to check person's mouth without punching them in the stomach.

Right, I forgot to add that to the to-do list. I'm going to have examine text say "They very clearly have something in their mouth" when appropriate.

**Comment by Anton-Kr** [comment]
> Right, I forgot to add that to the to-do list. I'm going to have examine text say "They very clearly have something in their mouth" when appropriate.

Imho I wouldn't make it THAT obvious as that would ruin the smuggling aspect of it. Just require a specific action to check.

**Comment by GreatBigFailure** [comment]
>Imho I wouldn't make it THAT obvious as that would ruin the smuggling aspect of it. Just require a specific action to check.

I mean, shift clicking the person would be that specific action. I'll make it require being next to the person, how about that?

**Comment by Anton-Kr** [comment]
That seems like a good compromise ya.

**Comment by GreatBigFailure** [comment]
@Anton-Kr Alright, the examine text is added:
- If their face is obscured then you don't get the examine text
- You need to be next to them. Considering changing it to 2 tiles, from the current 1
Now undown your thumb and grant me my dopamine

**Comment by Hinaichigo** [comment]
It's not that obvious if someone has a small object in their mouth. I guess if it's something like a fork or a nuke disc that would be noticeable, but things like matchboxes and cigarettes should be harder to detect. Security will always generally be on the adjacent tile when cuffing, strip-searching etc, so they would see the message most of the time. I'm for not having it be an examinable thing and you either need to forcibly open their mouth or otherwise make them spit it out to detect it, or get the hint by their mumbling speech.

**Comment by GreatBigFailure** [comment]
I don't think most people will notice the examine text unless they're specifically looking for it, if they even examine them at all. I'll keep it how it is for now and make changes if they're needed later. It's not meant to be a very strong way of smuggling things, more silly than anything. Mostly a benefit if they overlook it.

**Comment by Viciousaurus** [comment]
Now he mime can use their mouth for something

**Comment by Blithering** [comment]
> > There should be a way to check person's mouth without punching them in the stomach.
> 
> Empty handed disarm targeting the mouth maybe?

This action already has a unique interaction, not a very good one, it just silences them for a moment, but it's there.

**Comment by GreatBigFailure** [comment]
I'll see about making grab/disarm targeting the mouth be an alternate way to force someone to spit something out.
Though given the split on votes I might not put any more work into this since it might be pointless.

**Comment by wotisjanitor** [comment]
Add special rag interactions

**Comment by Eneocho** [comment]
One guy in the thread suggests to add a list of items that make an obvious mouth bulge, like the disk, grenades, etc.
All other items are not noticeable without doing the check.

**Comment by GreatBigFailure** [comment]
> Add special rag interactions

Unless they're particularly important any additional item-specific interactions will come in a later PR. I'd prefer to get the system working and added before getting fancy with fluff. Especially since I'm predicting a lot of bug fixing required post-live testing.
For that future PR though, what do you have in mind?

>All other items are not noticeable without doing the check.

I considered that and I'm not opposed to it. If it's the deal breaker here I'll do it but, at least at first, I planned on keeping the examine text as is. This isn't meant to be a particularly good method of smuggling things. It's a last ditch effort or for stupid gimmicks/emergent gameplay.

**Comment by Eneocho** [comment]
There was also a suggestion that vomiting while having something superglued in your mouth should suffocate you to death (vomit and stomach acid going into the lungs) or at least suffocate you a lot.

**Comment by SonixApache** [comment]
make it so the mob icon goes onto your glove slot when you're searching them for contraband

**Comment by GreatBigFailure** [comment]
> There was also a suggestion that vomiting while having something superglued in your mouth should suffocate you to death (vomit and stomach acid going into the lungs) or at least suffocate you a lot.

I increased how horrible vomiting while unable to spit out is. Gave it some significant but not unmanageable oxy damage, etc

**Comment by GreatBigFailure** [comment]
Changes, will work on the rest later.

- Vomiting with a glued item in your mouth is now more awful, and even more embarrassing
- Coughing and screaming (like when being tazed or during surgery) now makes you spit items out
- If an item has a sharp tip (screwdriver, knives, etc) it will do moderate damage to your head based on its sharpness when you put it in your mouth
- If an item has a sharp or serrated blade (scalpel, knife, etc) it will have a chance based on its sharpness to do sharpness based damage to you every tick while it's in your mouth
- If an item could be used to heat a beaker it will do 1 burn damage every tick. It might also set you on fire eventually.
- You can't eat food while you have something in your mouth. Trying to do so on harm intent has a small chance of making you choke on it
- Maybe actually made it work? I dunno!

**Comment by GreatBigFailure** [comment]
Added a way to force people to spit items out that doesn't involve punching:
- Grab intent, empty hand, targeting mouth
- If their mouth is blocked it fails
- Takes 1-2.5 seconds depending on their intent (help is 1 second, anything else is 2.5)
- Considering doing some bite damage to your hand if they're on harm intent with bite attack active, y/n?

**Comment by Eneocho** [comment]
> * Considering doing some bite damage to your hand if they're on harm intent with bite attack active, y/n?

Yes.

Also, add crowbar to mouth for brutality.

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -153,7 +153,9 @@
 				delayNextAttack(10)
 			if(INVOKE_EVENT(src, /event/uattack, "atom" = A)) //This returns 1 when doing an action intercept
 				return
-			UnarmedAttack(A, 1, params)
+			if(!biteCheck(A))
+				to_chat(world, "CLICK after bite check")
```
Comment (MadmanMartian) [inline code/_onclick/click.dm:157]: Leftover debug

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -268,8 +268,17 @@
 			return
 		//user.next_move = max(user.next_move+2,world.time + 2)
 	add_fingerprint(user)
-	if(can_pickup(user) && !user.put_in_active_hand(src))
-		forceMove(get_turf(user))
+	if(can_pickup(user))
+		if(ishuman(user))
+			to_chat(world, "Is human in pickup")
```
Comment (MadmanMartian) [inline code/game/objects/items.dm:273]: Leftover debug

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -268,8 +268,17 @@
...
+			var/mob/living/carbon/human/H = user
+			if(H.attack_type == ATTACK_BITE && H.can_bite(src))
+				to_chat(world, "attack bite and can bite")
```
Comment (MadmanMartian) [inline code/game/objects/items.dm:276]: Leftover debug

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -268,8 +268,17 @@
...
+				H.putItemInMouth(src)
+				to_chat(world, "after put in mouth PICKUP")
```
Comment (MadmanMartian) [inline code/game/objects/items.dm:278]: Leftover debug

File: code/game/objects/items/weapons/shard.dm [hunk 5]
```diff
@@ -42,6 +42,15 @@
 	..()
 	return
 
+/obj/item/weapon/shard/mouth_act(mob/user, mInitial = FALSE, mConstant = FALSE)
+	..()
+	if(mInitial)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.adjustBruteLossByPart(rand(20, 40), LIMB_HEAD, src)	//don't eat glass, dummy
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/shard.dm:50]: Isn't this within the damage bounds of decaptiation/pulping?

File: code/modules/mob/living/carbon/human/human_iteminmouth.dm [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+	name = "Spit Out"
+
+/datum/action/item_action/itemInMouth/proc/adjustForItem()
+	if(!target)
+		return FALSE
+	name = "Spit Out [target]"
+	desc = "Spits out \the [target] you have in your mouth"
+
+/datum/action/item_action/itemInMouth/Trigger()
+	to_chat(world, "Trigger start")
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/human/human_iteminmouth.dm:11]: Leftover debug

File: code/modules/mob/living/carbon/human/human_iteminmouth.dm [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+	if(ishuman(owner))
+		to_chat(world, "Trigger human")
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/human/human_iteminmouth.dm:13]: Leftover debug

**Comment by DriftingSkies** [comment]
There's a joke to be made here. I like this idea though, seems funny.

**Comment by WindowsErrors** [comment]
is this gonna be like that black guy in The First Purge who started the purge by attacking a drug dealer with a razor blade that he was keeping in his mouth

**Pull request closed without merging**